### PR TITLE
VEN-4 Add system parsing, support different types of return values, w…

### PR DESCRIPTION
**Pull request by bokimilinkovic**
Support paring all .go files in a single system(folder)

### CODE REVIEW

File: test-crawler/collector/collector.go [hunk 1]
```diff
@@ -88,3 +93,35 @@ func listTestFiles(root string, ignore []string) (files []string, err error) {
 
 	return nil, nil
 }
+
+// listGoFilesInFolder returns list of all .go files, excluding _test.go
+// Root folder will represent a system for which the methods will be crawled.
+func ListGoFilesInFolder(root string, ignore []string) (system string, files []string, err error) {
```
Comment (brdji) [inline test-crawler/collector/collector.go:99]: I assume `ignore` is meant to be used in the future?

**Comment by brdji** [comment]
How about the following test cases? 
- Crawl a folder which contains a `test` file
- Crawl a folder which includes no `go` files

File: test-crawler/crawler_test.go [hunk 2]
```diff
@@ -24,4 +23,22 @@ func TestExtractPublicMethodsFromFile(t *testing.T) {
...
+}
+
+func TestCrawlFolderForSystemMethods(t *testing.T) {
+	systemMethods, err := crawlFolderForSystemMethods("./mocks")
+	if err != nil {
+		t.Errorf("got error: %v", err.Error())
+	}
+
+	assert.Equal(t, systemMethods.System, "mocks")
+	assert.Equal(t, 6, len(systemMethods.Methods))
```
Comment (brdji) [inline test-crawler/crawler_test.go:43]: This unit test actually tests 2 things:
- Crawling a folder for files
- Extracting methods from each file

While the behavior `crawlFolderForSystemMethods` is not very complex, we should add more unit tests that target its parts.

When writing a unit test, we want it to test a single behavior, if possible. Therefore, one type of tests should test the method `ListGoFilesInFolder`, and include several tests for as many branches in the function as possible.

File: test-crawler/collector/collector.go (hunk 1 above)
Comment (bokimilinkovic) [reply test-crawler/collector/collector.go:99]: Yes, in the future that function will be expanded for different purposes, accepting what file types should be included, excluded, ignored, etc...

Comment (StefanGajic) [reply test-crawler/collector/collector.go:99]: Please add that to the comment for clarification.

File: test-crawler/crawler_test.go (hunk 2 above)
Comment (bokimilinkovic) [reply test-crawler/crawler_test.go:43]: Since we have nested function calls, I will extract testing of `ListGoFilesInFolder` in it's package module. If that function is well tested, no need to test single `crawlFolderForSystemMethods` since it is only combined call of `ListGoFilesInFolder` and `crawlSingleFileForFunctions`

Comment (brdji) [reply test-crawler/crawler_test.go:43]: Agreed. You can remove the test for `crawlFolderForSystemMethods`, and just write the tests for the two methods it encapsulates.
Of course, should this behavior change in the future, we will add new unit tests.

File: test-crawler/collector/collector_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		assert.NoError(t, err)
+		assert.Equal(t, system, "config")
+		assert.Equal(t, 2, len(files))
+	})
+
+	t.Run("Read golang files from mocks directory", func(t *testing.T) {
+		system, files, err := ListGoFilesInFolder("../mocks", nil)
+		assert.NoError(t, err)
+		assert.Equal(t, system, "mocks")
+		assert.Equal(t, 4, len(files))
```
Comment (brdji) [inline test-crawler/collector/collector_test.go:39]: This test is fragile - suppose someone adds a new file to the `mocks` or `config` folder, to test a feature that is unrelated to this one. Adding those files would cause this test to fail, even though the code changes did not affect this feature.

Use the `t.TempDir` approach, and create new temp files.

Comment (brdji) [reply test-crawler/collector/collector_test.go:39]: Another key point - the names of the tests tell us nothing about what they are covering. The tests should be named after the behavior they are actually testing.

Comment (bokimilinkovic) [reply test-crawler/collector/collector_test.go:39]: you are right, nice catch, thanks!

File: test-crawler/collector/collector_test.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+		err = ioutil.WriteFile(fmt.Sprintf("%s/b.go", tempDir), []byte("package main func main(){b:=2}"), 0644)
+		assert.NoError(t, err)
+
+		system, files, err := ListGoFilesInFolder(tempDir, nil)
+		assert.NoError(t, err)
+		splitedFolderName := strings.Split(tempDir, "/")
+		systemName := splitedFolderName[len(splitedFolderName)-1]
+		assert.Equal(t, system, systemName)
+		assert.Equal(t, 2, len(files))
+		os.Remove(tempDir)
```
Comment (brdji) [inline test-crawler/collector/collector_test.go:27]: Unnecessary: `t.TempDir()` should delete this directory. Please double-check.

Comment (bokimilinkovic) [reply test-crawler/collector/collector_test.go:27]: Yeah I also thought the same, but in the docs of ioutil package, it is sad:
`It is the caller's responsibility to remove the directory when no longer needed`

Comment (brdji) [reply test-crawler/collector/collector_test.go:27]: That is a completely different package. Compare the `ioutil` package:
image

With the `testing` package:
image

Comment (bokimilinkovic) [reply test-crawler/collector/collector_test.go:27]: Yeah that's correct. package testing is used for folder creation, and ioutil for writing into files. I will remove it
